### PR TITLE
ISPN-15961 JDBC_PING with server datasource

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-15961

I renamed  `TransportSecurityConfiguration` to `ServerTransportConfiguration` and moved it to `ServerConfiguration` as it is much more appropriate there. This breaks backwards compatibility for those extending the server, but I don't think that's  an issue.

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/xml/server-transport-security.xml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <server xmlns="urn:infinispan:server:{schemaversion}">
+  <datasource name=""
```
Comment (jackie-linz) [inline documentation/src/main/asciidoc/topics/xml/server-transport-security.xml:2]: invalid XML?

Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/xml/server-transport-security.xml]: Yeah :) I fixed it

**Review by pruivo — COMMENTED** [review]
LGTM. I only have minor comments. I'll test it out locally too.

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
The native compilation is failing @tristantarrant  :scream:

**Comment by tristantarrant** [comment]
@pruivo it's ok now

**Comment by pruivo** [comment]
merged! thanks @tristantarrant !

**Pull request merged**